kibana-reports/public/components/context_menu/context_menu.js

```diff
@@ -234,12 +234,34 @@ $(function () {
   locationHashChanged();
 });
 
+const isDiscoverNavMenu = (navMenu) => {
+  return navMenu[0].children.length === 5 && 
```

Comment:
why is it always 5, 4, or 3. I feel it's kind of fragile to used `== (a fixed number)`

Reply:
It checks how many items are in the nav menu depending on what kind of report source it is- I can extract them to constants too

Reply:
can we remove this length check, since you have the title check？Or can we just use `length >= 0` or `length >= 1`? What's the downside you see

Reply:
That was the old check, which resulted in the issue. Basically it would not add the Reporting menu if the nav menu had length greater than 1, which is why the Reporting menu would not show up. But we need to not only check that the Nav menu is populated, but that it has the correct number of items

Reply:
a better way might be to add an id to reports menu element, loop through navMenu and check for the id to determine whether reports menu is added. probably a todo item

Reply:
+1 - I think that's a good idea but it doesn't sound like a quick fix